release: require macOS 10.10+

fanquake opened the pull request:
Closes #13362

d99abfddb0c8f2111340a6127e77cc686e0043d8
This workaround should no longer be required, as it should have only been in use when compiled with the 10.7 SDK, which we haven't been building with for a while now.

5bc5ae30982a0f0f6a9804b05d99434af770c724
The bugreport linked with this code is for an unrelated? issue, however from what I can tell the correct QTBUG is this one https://bugreports.qt.io/browse/QTBUG-20880. Reading though the discussion there, it seems that the way progress bars are animated changed in macOS 10.10. 
Qt was patched [here (5.5+)](https://codereview.qt-project.org/#/c/112379/):
> Disable progress bar animations on 10.10 Yosemite and higher - the native style does not animate them any more. Keep the indeterminate progress bar animation.

Given all of that, I don't think this is worth keeping around, as it would seem to only be useful in the case that a macOS user is compiling with a Qt < 5.5. That should be pretty unlikely, as we don't support downloaded Qt binaries, and brew currently provides [5.11.1](https://github.com/Homebrew/homebrew-core/blob/571b46213c70ca1573da6d0425b0bd6df34961ee/Formula/qt.rb).

fanquake committed:
depends: set OSX_MIN_VERSION to 10.10

fanquake committed:
release: bump minimum required macOS to 10.10

fanquake committed:
doc: mention that macOS 10.10 is now required

fanquake committed:
gui: remove SubstituteFonts

fanquake committed:
gui: remove macOS ProgressBar workaround

Empact commented:
0f57c60 could be a scripted-diff

practicalswift commented:
Concept ACK

theuni commented on src/util.cpp:
```diff
@@ -719,8 +719,8 @@ fs::path GetDefaultDataDir()
         pathRet = fs::path(pszHome);
-#ifdef MAC_OSX
-    // Mac
+#ifdef __APPLE__
```
Do we for sure mean "Apple" in all of these changes and not just "macOS"? This would affect iOS as well, which seems likely incorrect here. I suspect that we want to keep using MAC_OSX (which we define ourselves) so that we can differentiate later if needed.

jonasschnelli replied:
Agree with @theuni

fanquake replied:
I've dropped this change and replaced it with a scripted-diff that converts __APPLE__ usage to MAC_OSX.

theuni commented on depends/README.md:
```diff
@@ -22,7 +22,7 @@ Common `host-platform-triplets` for cross compilation are:
 - `i686-w64-mingw32` for Win32
 - `x86_64-w64-mingw32` for Win64
-- `x86_64-apple-darwin11` for macOS
+- `x86_64-apple-darwin14` for macOS
```
Does this change anything?

MarcoFalke replied:
Huh, how does this even compile on travis/gitian, given that support for 10.7 has been removed?

theuni replied:
I've never seen a case where this changes anything in practice. As a test, I just had a look at gcc's configure.ac though, and it at least checks for >=10 in a few places. So I guess we should keep these in sync to be on the safe side.

fanquake replied:
Updated in travis.yml and gitian-osx.yml

jonasschnelli commented on src/qt/guiutil.h:
```diff
@@ -233,19 +230,7 @@ namespace GUIUtil
         void mouseReleaseEvent(QMouseEvent *event);
     };
 
-#if defined(Q_OS_MAC)
```
Has someone verified that this BUG no longer appears on OSX 10.10+?

fanquake replied:
@jonasschnelli I haven't been able to recreate it yet, will test further though.

kallewoof replied:
The new URL for the bug appears to be https://bugreports.qt.io/browse/QTBUG-15631 but that bug seems unrelated to the comment.

fanquake replied:
@kallewoof I mentioned in the PR description, from what I can tell the correct link is https://bugreports.qt.io/browse/QTBUG-20880

jonasschnelli reviewed (commented):
Concept ACK.
Changes look good.
Does someone has a 10.10 build environment? I guess would be great to test on 10.9 (failcase-test) and on 10.10.

fanquake committed:
scripted-diff: prefer MAC_OSX over __APPLE__

-BEGIN VERIFY SCRIPT-
sed -i 's/__APPLE__/MAC_OSX/g' src/compat/byteswap.h src/util.cpp
-END VERIFY SCRIPT-

MarcoFalke commented:
Appears to time out, but builds on my travis when merged with #13634

MarcoFalke commented:
Should the darwin11 also be updated when `make download-osx` in depends?

Sjors commented:
tACK 5eb615c, modulo #13752 and the `depends/Makefile` darwin11 reference @MarcoFalke pointed out.

I also built depends, ran functional test with that, then ran regular tests without depends (see #13750). I switched GUI to Mandarin, looks nice. I didn't check if the progress bar uses CPU when disconnected.

theuni commented:
utACK after the Makefile change.

fanquake commented:
Added the makefile change to 26b15df

kallewoof commented:
utACK 3828a79

I assume the `__APPLE__` define is still available, as it is still used in

https://github.com/bitcoin/bitcoin/blob/1211b15bf6c0b2904d90b96a9b3834c5cb9e7b4e/src/tinyformat.h#L157

Sjors commented:
@fanquake what was the rationale for using `MAC_OSX`? I can't find any documentation about it.

This (fairly) old blog post recommends a different strategy: http://nadeausoftware.com/articles/2012/01/c_c_tip_how_use_compiler_predefined_macros_detect_operating_system#OSXiOSandDarwin
